### PR TITLE
[lexical-playground] CI: fix flaky collab test

### LINKED ISSUE #6241: Stabilise CI: resolve flaky e2e tests
disable/fix e2e tests that are not deterministic, aka flaky

### PR DESCRIPTION
**Pull request by potatowagon**
## Description
https://github.com/facebook/lexical/actions/runs/9294132717/job/25578693971?pr=6201

^ this e2e collab test fails occasionally on the CI due to flakiness. the output is an extra `<p></br><p>`

add a callback to modify actual html. this callback is useful to adjust for flaky html outputs

Closes #6241

## Test plan
```
npm run start & npm run test-e2e-collab-chromium
```
 ✓  1 [chromium] › packages/lexical-playground/__tests__/e2e/Toolbar.spec.mjs:37:8 › Toolbar › Insert image caption + table (2.5s)

### CODE REVIEW

File: packages/lexical-playground/__tests__/e2e/Toolbar.spec.mjs [hunk 1]
```diff
@@ -99,6 +99,25 @@ test.describe('Toolbar', () => {
         ignoreClasses: true,
         ignoreInlineStyles: true,
       },
+      (actualHtml) =>
+        // flaky fix: remove the extra <p><br /></p> that appears occasionally in CI runs
```
Comment (Sahejkm) [inline packages/lexical-playground/__tests__/e2e/Toolbar.spec.mjs:103]: do we want to add a check for collab if flakyness is for collab mode only ?

**Review by Sahejkm — APPROVED** [review]

File: packages/lexical-playground/__tests__/e2e/Toolbar.spec.mjs (hunk 1 above)
Comment (potatowagon) [reply packages/lexical-playground/__tests__/e2e/Toolbar.spec.mjs:103]: afaik these extra chars only appear for collab, tho if it happens to appear for non collab, no harm helping to remove the xtra chars so the test is not flaky

**Comment by Sahejkm** [comment]
i think we don't need to add label if the PR is approved ald, it cancelled the run https://github.com/facebook/lexical/actions/runs/9346276629/usage and restarted a new one :(

**Comment by potatowagon** [comment]
> i think we don't need to add label if the PR is approved ald, it cancelled the run https://github.com/facebook/lexical/actions/runs/9346276629/usage and restarted a new one :(

sorry, perhaps i forgot to refresh the page, i added the tag when i saw the e2e tests were skipped

**Pull request merged**